weather/weather.go

```diff
@@ -131,18 +154,30 @@ func (p *Parser) Parse(values url.Values) {
 
 	updateGauge(p.battery.WithLabelValues(p.name, "outdoor"))(parseValue("battout"))
 	updateGauge(p.battery.WithLabelValues(p.name, "indoor"))(parseValue("battin"))
+	updateGauge(p.battery.WithLabelValues(p.name, "lightning"))(parseValue("batt_lightning"))
 	updateGauge(p.humidity.WithLabelValues(p.name, "indoor"))(parseValue("humidityin"))
 	updateGauge(p.barometer.WithLabelValues(p.name, "relative"))(parseValue("baromrelin"))
 	updateGauge(p.barometer.WithLabelValues(p.name, "absolute"))(parseValue("baromabsin"))
-	updateGauge(p.windDir.WithLabelValues(p.name))(parseValue("winddir"))
+	updateGauge(p.windDir.WithLabelValues(p.name, "current"))(parseValue("winddir"))
+	updateGauge(p.windDir.WithLabelValues(p.name, "avg10m"))(parseValue("winddir_avg10m"))
 	updateGauge(p.windSpeedMph.WithLabelValues(p.name, "gusts"))(parseValue("windgustmph"))
 	updateGauge(p.solarRadiation.WithLabelValues(p.name))(parseValue("solarradiation"))
+	updateGauge(p.rainIn.WithLabelValues(p.name, "hourly"))(parseValue("hourlyrainin"))
 	updateGauge(p.rainIn.WithLabelValues(p.name, "daily"))(parseValue("dailyrainin"))
 	updateGauge(p.rainIn.WithLabelValues(p.name, "weekly"))(parseValue("weeklyrainin"))
 	updateGauge(p.rainIn.WithLabelValues(p.name, "monthly"))(parseValue("monthlyrainin"))
 	updateGauge(p.rainIn.WithLabelValues(p.name, "yearly"))(parseValue("yearlyrainin"))
+	updateGauge(p.rainIn.WithLabelValues(p.name, "total"))(parseValue("totalrainin"))
 	updateGauge(p.rainIn.WithLabelValues(p.name, "event"))(parseValue("eventrainin"))
 	updateGauge(p.ultraviolet.WithLabelValues(p.name))(parseValue("uv"))
+	updateGauge(p.lightning.WithLabelValues(p.name, "day", "total"))(parseValue("lightning_day"))
+	updateGauge(p.lightning.WithLabelValues(p.name, "last", "distance"))(parseValue("lightning_distance"))
+	updateGauge(p.lightning.WithLabelValues(p.name, "last", "time"))(parseValue("lightning_time"))
+
+	stationType, station_err := parseString("stationtype")
+	if err == station_err {
+		updateGauge(p.stationtype.WithLabelValues(p.name, stationType))(float64(1), nil)
+	}
```

Comment:
What use do you get out of a metric with a label of your station type? Is it useful somehow?

You have an error here; `if err == station_err {` should be `if station_err != nil {`